node/actors/artifacts.go

```diff
@@ -24,24 +28,57 @@ type ArtifactReference struct {
 	Size int
 }
 
+func extractTag(location *url.URL) string {
+	return location.Fragment
+}
+
 // Obtains an artifact from the specified location. If the indicated location allows for
 // differentiation using tags (e.g. OCI, Object Store), then the supplied tag will be used,
 // otherwise it will be ignored
-func GetArtifact(name string, tag string, location *url.URL) (*ArtifactReference, error) {
+func getArtifact(name string, uri string) (*ArtifactReference, error) {
+	location, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+
 	switch location.Scheme {
 	case SchemeFile:
 		return cacheFile(name, location)
 	case SchemeNATS:
-		return cacheObjectStoreArtifact(name, tag, location)
+		return cacheObjectStoreArtifact(name, extractTag(location), location)
 	case SchemeOCI:
-		return cacheOciArtifact(name, tag, location)
+		return cacheOciArtifact(name, extractTag(location), location)
 	}
 	return nil, nil
 }
 
 func cacheFile(name string, location *url.URL) (*ArtifactReference, error) {
-	// TODO: Implement
-	return nil, nil
+	filePath, tag := "", ""
+	if strings.Contains(location.Path, ":") {
+		sPath := strings.Split(location.Path, ":")
```

Comment:
for safety's sake, we should probably use `SplitN(s, ":", 1)`, which will avoid crashing things if the tag actually has a `:` in it (we should block that.. but safety iz gud)